### PR TITLE
Add list of useful references for SIG-API reviews

### PR DESCRIPTION
**Pull request by dthaler**
Signed-off-by: Dave Thaler <dthaler@ntdev.microsoft.com>

### CODE REVIEW

File: docs/Community/sig-api/README.md [hunk 1]
```diff
@@ -18,14 +20,63 @@ Dave Thaler ([dthaler](https://github.com/dthaler))
...
+  that the memory really is available.
+  Hence this reference should be consulted for discussion of "standard" APIs.
+
+* [Windows API documentation](https://docs.microsoft.com/en-us/windows/win32/api/):
+  Windows is not always compliant to the POSIX API or C/C++ standards.
+  For example, Windows requires [`closesocket()`](https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-closesocket)
+  to close a socket rather than the POSIX `close()` API.
+  Hence this reference should be consulted for discussion of "standard" APIs.
+
+* (Microsoft-internal only) [Win32 API Design Guidelines](https://osgwiki.com/wiki/Win32_API_Design_Guidelines):
```
Comment (CodeMonkeyLeet) [inline docs/Community/sig-api/README.md:79]: Can we instead refer to public API design standards? Particularly as Intel and external contributors become more common, this suggests a kind of double-standard. Many of these exist for web APIs, but I don't know of any for C/C++ APIs.

Alternatively, consider abstracting whatever additional guidelines you wish to enforce in OE SDK into the OE SDK API design guidelines. (e.g. the recommendation against the call-twice APIs)

Comment (dthaler) [reply docs/Community/sig-api/README.md:79]: Completely agree, but I haven't found public ones yet.   I can ask those who maintain the link above if there is a public equivalent, but is it ok to address this comment in a separate PR once one is found or created?

Comment (CodeMonkeyLeet) [reply docs/Community/sig-api/README.md:79]: I would remove the reference to Microsoft internal documents from this PR and add it in a separate PR if there is not already an existing public version of the doc. I don't think it's appropriate to reference information in the public docs that are only accessible to the MS maintainers.

Comment (dthaler) [reply docs/Community/sig-api/README.md]: Done

**Comment by dthaler** [comment]
bors try

**Review by CodeMonkeyLeet — APPROVED** [review]
Thanks for putting this together!

**Comment by BRMcLaren** [comment]
bors try

**Review by BRMcLaren — APPROVED** [review]
LGTM thanks for putting this together!

**Pull request merged**